build.gradle

```diff
@@ -203,9 +197,27 @@ configure(javaProjects) {
         }
     }
 
+    task removeGeneratedSources(type: Delete) {
+        generatedSourcesDir.deleteDir()
+    }
+
+    task removeIdeaOutDirectory(type: Delete) {
+        file("${projectDir}/out").deleteDir()
+    }
+
+    clean {
+        dependsOn(removeIdeaOutDirectory)
+    }
+
     compileJava {
+        doFirst {
+            generatedSourcesDir.mkdirs()
+            generatedSourcesJavaDir.mkdirs()
+        }
+        dependsOn(removeGeneratedSources)
```

Comment:
This will delete and regen every time we compile, so all downstream tasks of compile will never be up to date

Reply:
Seems to work better than before when the downstream worked every time. Since the hash of the generated files are always the same (provided no entity class changes) the outputs are marked as the same and the tasks don't run.